src/core.js

```diff
@@ -104,8 +104,12 @@ async function sync(cloneUrl, localDirectory, {ref, checkout}) {
   }
 
   await git.cwd(localDirectory);
+
+  await git.removeRemote(cloneName).catch((error) => {}); // eslint-disable-line no-unused-vars
```

Comment:
this cloneUrl is specific to the github-token so if a token is revoked, then all the old remotes with invalid tokens are still there and will fail on the fetch.

always doing a `.removeRemote()` right here means the cloneUrl provided is always used.